### ISSUE TITLE
Correlation between batch_size and num_interations

### ISSUE THREAD

**Issue by maxnk** [jcjohnson/fast-neural-style#80]
I'm trying to find an optimal way to train a new models and came across the aforementioned parameters.

Let's say, I have a **GTX970/4GB** and I use
`-batch_size 4 -num_iterations 40000`
These are default values and I receive pretty good results for most models I tried to train with them.

Would it be correct to assume that I'll receive the same results on **K80/12GB** with the following params?:
`-batch_size 16 -num_iterations 10000`

Or, maybe there are some other parameters which worth to be played with?
And if yes, then how do they correlate between each other?

**Comment by jcjohnson** [comment]
If you increase the batch size you may be able to get faster convergence by also increasing the learning rate; in addition to the default `1e-3` I would also try `-learning_rate 2e-3` and `-learning_rate 4e-3`.

However it's impossible to say for sure that you'd get the same results; these are just general heuristics and may not be applicable in all cases.

**Comment by maxnk** [comment]
Okay, thanks for your reply, so there is a field for experiments :)

Btw, do you have any empiric-like metrics for defining the successful convergence, to stop training programmatically based on them?
For example, "loss shouldn't change by more than 0.5% during last 100 iterations".
